src/runtime/google-analytics-event-listener.js

```diff
@@ -65,16 +75,61 @@ export class GoogleAnalyticsEventListener {
     }
 
     // TODO(b/234825847): Remove it once universal analytics is deprecated in 2023.
-    if (typeof this.win_.ga === 'function') {
-      this.win_.ga('send', 'event', gaEvent);
+    const ga = this.win_.ga || null;
+    if (isFunction(ga)) {
+      ga(
+        'send',
+        'event',
+        Object.assign(
+          {},
+          gaEvent,
+          !!analyticsParams?.eventCategory
+            ? {
+                eventCategory: analyticsParams.eventCategory,
+              }
+            : null,
+          !!analyticsParams?.eventLabel
+            ? {eventLabel: analyticsParams.eventLabel}
+            : null
+        )
+      );
     }
 
-    if (typeof this.win_.gtag === 'function') {
-      this.win_.gtag('event', gaEvent.eventAction, {
```

Comment:
Then you can split `gaEvent.eventAction` and the rest.

```
const {eventAction, ...gtagEventParam} = gaEvent;
this.win_.gtag('event', eventAction, gtagEventParam);
```

Reply:
Sorry, can you be more explicit here? How does `gtagEventParam` have the required attributes defined? Another step we have to do is convert the attributes from `snake_case` to `camelCase`.

Reply:
Oh, I didn't notice they are based on different cases styles. In that case, I think it would make more sense for the callers to set the attributes in `analyticsParams` in snake_case when it passes it in (as `ga` will be deprecated next year).

Then we can do some manual overwites, e.g., 
```
gaEvent = {
   ...gaEvent,
   eventCategory: analyticsParams.event_category || gaEvent.eventCategory,
   eventLabel: analyticsParams.event_label || gaEvent. eventLabel,
}
```

For `gtag`, we can do the manual conversion (as it was doing) and then overwrite.

Reply:
how do these changes look?

Reply:
Looks great!